src/xSTIR/cSTIR/cstir.cpp

```diff
@@ -1249,6 +1266,36 @@ cSTIR_computeObjectiveFunctionGradientNotDivided(void* ptr_f, void* ptr_i, int s
 	CATCH;
 }
 
+extern "C"
+void*
+cSTIR_objectiveFunctionAccumulateHessianTimesInput
+    (void* ptr_fun, void* ptr_est, void* ptr_inp, int subset, void* ptr_out)
+{
+	try {
+		ObjectiveFunction3DF& fun = objectFromHandle<ObjectiveFunction3DF>(ptr_fun);
+		STIRImageData& est = objectFromHandle<STIRImageData>(ptr_est);
+		STIRImageData& inp = objectFromHandle<STIRImageData>(ptr_inp);
+		STIRImageData& out = objectFromHandle<STIRImageData>(ptr_out);
+		Image3DF& curr_est = est.data();
+		Image3DF& input    = inp.data();
+		Image3DF& output   = out.data();
+		if (subset >= 0)
+			fun.accumulate_sub_Hessian_times_input(output, curr_est, input, subset);
+		else {
+			int nsub = fun.get_num_subsets();
+			output.fill(0.0);
```

Comment:
if we're accumulating in SIRF (which I don't think we should), then this `fill` is wrong. On the other hand, if we're not accumulating, then you'll need it also in the case `subset>=0`

Reply:
moved the fill above `if` for now - will get rid of it when we switch to `multiply_with_Hessian`